react/src/App.tsx

```diff
@@ -34,6 +35,7 @@ function BaseApp(props: { darkMode: boolean; toggleDarkMode: () => void }) {
                 const loginInfo = await result.json();
                 setUsername(loginInfo.username);
                 setLastLoginTime(loginInfo.last_login);
+                setIsAdmin(loginInfo.is_admin);
```

Comment:
This needs to be passed down to LoginForm too or the admin page will continue to be rendered/not rendered until a full refresh.

Reply:
Right, in case they re-sign in, good catch!